### ISSUE TITLE
dotdict expression not resolved

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#89]
JSON file:

```
// composite data structure
"params" : [{"dict_1_key_1" : "dict_1_key_1 value",
             "dict_1_key_2" : ["dict_1_key_2 value 1", "dict_1_key_2 value 2"]},
            //
            {"dict_2_key_1" : "dict_2_key_1 value",
             "dict_2_key_2" : {"dict_2_A_key_1" : "dict_2_A_key_1 value", "dict_2_A_key_2" : "dict_2_A_key_2 value"}}],
//
// access to single elements of composite data structure
//
// a) standard notation
"dict_1_key_2_value_2_standard" : ${params}[0]['dict_1_key_2'][1]
// b) dotdict notation
"dict_1_key_2_value_2_dotdict" : ${params.0.dict_1_key_2.1}
```

Outcome:

a) standard notation works
b) dotdict notation causes: `The variable '${params}['0']['dict_1_key_2']['1']' is not available`

Seems that also indices are assumed to be key names as strings.

But the element after a dot must be interpreted depending on the data type (list or dictionary) of the element before the dot.

I would expect that both notations result in the same value: `'dict_1_key_2 value 2'`

**Comment by test-fullautomation** [comment]
Hi Son,

this is actually tricky. In case of an array the `0` in `${params.0.dict_1_key_2.1}` need to be considered as integer, if it is a data structure it can be a string. Unfortunately the dot syntax is ambiguous here.
It should work independent of the data type to be user friendly.

Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas, Hello Holger,

This enhancement was implemented in the [commit 45f79386](https://github.com/test-fullautomation/python-jsonpreprocessor/commit/45f793867fed297705633daef5ab09c809792c7b) on stabi branch ([namsonx/task/stabi_branch](https://github.com/test-fullautomation/python-jsonpreprocessor/tree/namsonx/task/stabi_branch))

Thank you,
Son

**Comment by HolQue** [comment]
Retest result:

dotdict notation works now.

Issue can be closed.
